src/game.cpp

```diff
@@ -3178,6 +3178,8 @@ bool game::try_get_right_click_action( action_id &act, const tripoint &mouse_tar
         return false;
     }
 
+    const bool is_adjacent = std::abs(mouse_target.x - u.posx()) <= 1 && std::abs(mouse_target.y - u.posy()) <= 1;
+    const bool is_self = std::abs(mouse_target.x - u.posx()) <= 0 && std::abs(mouse_target.y - u.posy()) <= 0;
```

Comment:
That probably won't be a safe assumption once z-levels are added. I'd imagine you'd be able to offset your view by z-levels, just as you can offset it by x and y currently.